pkg/controllers/csi/metadata/sqlite_gorm_client.go

```diff
@@ -0,0 +1,197 @@
+package metadata
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type DBAccess interface {
```

Comment:
How are we handling the reading of soft-deleted entries?
Do our `Read...` functions consider this?

So, as of right now, the database is not used that much for the GC logic.
- Currently, we check what is on the filesystem, try to match a DB entry, and remove stuff the database doesn't know about.
   - This is bad because if we have to change the location of something, the logic for the GC has to know the old and new locations, which is rather annoying.
- In the future, we would like to do the other way around, so look at the database and decide what to remove from the filesystem.
  - To do this, we will need separate calls to look at what was soft-deleted. 
  - We can add those functions when we look more deeply at how the garbage collection should work.

Reply:
No our Read funcs only work with live data.
imho better to have a kind of "Cleanup" function that removes all entries older than X or something like this.

As I do not understand the requirements of GC clearly I wanted to get the API in and refactor when reworking the GC of the CSI driver.

Reply:
normally I would also separate APIs/interfaces by use-case, so I like your idea to have a different interface for GC.